apps/web/components/booking/BookingDescription.tsx

```diff
@@ -40,6 +42,18 @@ interface Props {
 const BookingDescription: FC<Props> = (props) => {
   const { profile, eventType, isBookingPage = false, children } = props;
   const { t } = useLocale();
+  const { duration, setQuery: setDuration } = useRouterQuery("duration");
+  useEffect(() => {
+    if (eventType.metadata?.multipleDuration !== undefined) {
+      if (!duration) {
+        setDuration(eventType.length);
+      } else {
+        if (!eventType.metadata?.multipleDuration.includes(Number(duration))) {
+          setDuration(eventType.length);
+        }
+      }
+    }
+  }, []);
```

Comment:
This is to control the duration coming from the query parameter, if it's a valid one let it pass, otherwise use the default one stored in the same "length" property of the eventType.